### ISSUE TITLE
Code generation: ensure that a model is runnable

### ISSUE THREAD

**Issue by agarny** [cellml/libcellml#493]
For code generation to work, a model must be valid, i.e. a call to `Validator::validateModel()` must return without any errors.

However, a valid model is no guarantee that code can be generated from it. Indeed, `Validator::validateModel()` makes sure that a model is valid with regards to the CellML specifications, but **not** whether a model is runnable.

So, as part of processing a model, our generator should check that the model is indeed runnable, and return errors if it is not.

An example of a valid model, but not runnable model, can be found in issue #492 (second example) where we have a model that has a variable that is both initialised and computed. Indeed, the only case where this is allowed is if the variable is computed through an ODE. However, in that example, the variable is computed through an algebraic equation.

In summary, this issue is about going through our `Generator` class and removing any "assumption" we make with regards to a model being runnable.

**Comment by kerimoyle** [comment]
Just to note that in addition to the two in #492 there is also the malformed MathML (as in #459) which segfaults too...

**Comment by agarny** [comment]
Closing this issue in favour of issue #499.
